sensor_msgs/msg/NavSatStatus.msg

```diff
@@ -4,12 +4,13 @@
 # type and the last time differential corrections were received.  A
 # fix is valid when status >= STATUS_FIX.
 
+int8 STATUS_UNKNOWN = -2        # status is not yet set
```

Comment:
This makes sense to me to have an uninitialized state as the default.

Reply:
I view it as a safety problem that the default state is "healthy". Because ROSIDL supports a default constructor, users of NavSatFix are not required to fill this out. Thus, if they take shortcuts, only fit out the lat-long fields, and lose GPS signal,they now report the status as having a fix even though the data is invalid and might not realize it.

Reply:
Doesn't defaulting to `STATUS_NO_FIX` fix the problem?

Reply:
You could also default the latitude and longitude to NAN to avoid accidental visits to Null Island. https://en.wikipedia.org/wiki/Null_Island

Reply:
> Doesn't defaulting to `STATUS_NO_FIX` fix the problem?

Yes, but adding a new value makes it exactly clear between these two situations:
1. The ROS driver has started but it has not received a fix status message from the GPS =>`STATUS_UNKNOWN`
2. The ROS driver has communicated with the GPS, and the GPS said it has no fix => `STATUS_NO_FIX`

I'd vote for keeping these as distinct as one means you might have a wiring or baud issue, and the other means that you just need to wait for it to warm up or get a better view of the sky.

Reply:
I am a fan of setting the latitude and longitude to NAN as default (assuming the ROS IDL lets us do that).

Reply:
> You could also default the latitude and longitude to NAN to avoid accidental visits to Null Island. https://en.wikipedia.org/wiki/Null_Island

This is not currently possible. See my docs PR which explains what's currently possible with `NaN`.
https://github.com/ros2/ros2_documentation/pull/4210

I am working on adding support for NaN defaults and NaN constants, but it's not clear whether it will make the Jazzy release.

Reply:
Adding in the NaN logic is reasonable, but I think that should be a sanity check of the data, more than the canonical way to know what the status is. Checking all field values for NaN can be very expensive versus being able to quickly check an enumeration and stop processing data early.

Reply:
I would be more of a NaN fan if tooling supported it better, but there's currently no way to to set a NaN value in DDS IDL, nor is it supported in JSON. See here for implementation notes. I'd prefer not hold up this PR which was intended as a bugfix with other things like NaN support.